### ISSUE TITLE
♻️ Update and combine the accordion and collapsible components

### ISSUE THREAD

**Issue by JoshBeveridge** [GCTC-NTGC/gc-digital-talent#14764]
### ♻️ Refactor

[Figma link](https://www.figma.com/design/guHeIIh8dqFVCks310Wv0G/Component-library?node-id=5-20)

Our existing `Accordion` and `Collapsible` component have known accessibility issues and are being used in inconsistent ways. We want to propose a single `Disclosure` component that replaces both and addresses any existing issues.

### 🕵️ Details

The aim with this refactor is to simplify and consolidate as the new `Disclosure` component to create more consistent usage and an improved accessibility experience.

<img width="2418" height="1400" alt="Image" src="https://github.com/user-attachments/assets/2a9cd453-f918-4385-bed8-963e0efe0c44" />

### 🙋‍♀️ Proposed Solution

For details on the following lists, please refer to the visuals and annotations provided in Figma.

#### Title styles

When the component is used as FAQ-style questions or as contextual definition lists, it should use the default `body` title style. When the component is used as a section or tool, it should use the `heading` title style. Note that the latter only has a single font-size and weight - it should not visually adapt to the `<hx>` element passed to the `title` slot.

#### Slots

The following content should be passed into slots and adjust the styling accordingly:

* `title`: Required, accepts `<hx>` or `<p>` tags, shows the disclosure's title, size and bolding should follow the title styles described above
* `actions`: Optional, accepts a list of up to 2 icon-only `<button>` tags
* `subtitle`: Optional, accepts unformatted text, shows the disclosure's subtitle in the summary area
* `metadata`: Optional, accepts a list of objects that fall into 1 of 4 categories: unformatted text, actions (`<a>` and `<button>` tags), `Chip` components, or `Status` components
* `content`: Required, accepts HTML and provides the disclosure with body content

#### Props

The following content should be controlled using props:

* `expanded`: Allows the disclosure to be expanded by default
* `divider`: Toggles the presence of a divider on the top edge of the component
* `horizontalPadding`: Toggles padding on the left and right
* `verticalPadding`: Toggles padding on the top and bottom
* `striped`: Provides a candy stripe background to the component

#### Disclosure groups

The component can be grouped in 2 visual styles: inline and card. Inline groups are more streamlined and used in the context of other copy where definition lists or supplementary contextual information is helpful. Card groups should be used to highlight important information like FAQs or requirement lists. In both cases, the group should accept an optional `footer` slot that can be passed formatted text. All disclosure groups should also be accompanied by an "Expand/Collapse all" action.

<img width="4164" height="2558" alt="Image" src="https://github.com/user-attachments/assets/2f388a98-dccc-43c0-9184-5e6d24dbe179" />

#### Business requirements
 - [ ] Consolidate the `Accordion` and `Collapsible` components
 - [ ] All items in the https://github.com/GCTC-NTGC/gc-digital-talent/milestone/53 milestone are addressed

#### Design requirements
- [ ] Light and dark mode have both been adapted according to Figma
- [ ] Narrow and wide screen size variations have been adapted according to Figma
- [ ] The component has 2 title styles
- [ ] The component has 2 required and 3 optional slots
- [ ] The component has 2 disclosure states and can be set to either by default
- [ ] The component has an optional divider
- [ ] The component has 2 padding options
- [ ] The component has an optional candy-striping style
- [ ] The component has 2 group styles

#### Development requirements
- [ ] TBD

#### Wrap-up
- [ ] A Storybook story has been created to include a component page showcasing all criteria outlined in this ticket
- [ ] Tests have been written to ensure visual diffs are caught and interactivity is accounted for
- [ ] The component has been Fable tested to account for all relevant WCAG 2.1 criteria that apply to disclosure elements
- [ ] A designer has reviewed the Storybook story and approved the component

**Comment by petertgiles** [comment]
Reviewed in refinement:
- Peter and Eric will do a review of the new design and provide any feedback.

**Comment by petertgiles** [comment]
Thanks for posting this Josh.  I've taken a look through this and had a few comments.
1. There is currently no Collapsible component.  So this scope of this issue would be just to adapt the current Accordion component into a new collection of disclosure components.
2. To break this down, I suggest something like:
    1. create new disclosure components with storybook
    2. have them Fable tested
    3. replace existing accordions
    4. remove accordion component
If Fable testing in Storybook is insufficient we could replace a small number of accordions in the app and ask them to test that.
3. It seems to me like there is a pretty big difference about how the disclosure heading looks visually and to AT.  I'm a little worried it could be difficult to implement.
4. Just flagging that in the metadata and action buttons, we'll need to be able to support dialogs which maintain focus position when closed.

**Comment by petertgiles** [comment]
Moved this to the "Component library" feature.
